clang/lib/Sema/SemaChecking.cpp

```diff
@@ -5406,6 +5408,103 @@ ExprResult Sema::ConvertVectorExpr(Expr *E, TypeSourceInfo *TInfo,
                                    RParenLoc, CurFPFeatureOverrides());
 }
 
+ExprResult Sema::BuiltinInvoke(CallExpr *TheCall) {
+  SourceLocation Loc = TheCall->getBeginLoc();
+  auto Args = MutableArrayRef(TheCall->getArgs(), TheCall->getNumArgs());
+  assert(llvm::none_of(Args,
+                       [](Expr *Arg) { return Arg->isTypeDependent(); }));
+
+  if (Args.size() == 0) {
```

Comment:
Instead of all of these separate diagnostics, is there a reason we can't just use `Sema::checkArgCountAtLeast` and friends?   This one looks like a 'at least 1'.  5427 looks like `Sema::checkArgCountAtMost`, and 5440 `Sema::checkArgCount`.

WDYT?

Reply:
AFAICT `checkArgCountAtLeast` actually prints "expected x, have y". IMO the "at least" here is quite important, since the number of required arguments changes depending on the first argument. I guess I could introduce a new helper; not sure that this is the patch to do it though. I've replaced the 5440 one.

Reply:
Ah, thats unfortunate.  But yes, I think that seems to be the way it is.  Its a little odd we don't actually print the 'right' thing for the 'at least' function, but I see this is far from the first.  SGMT.